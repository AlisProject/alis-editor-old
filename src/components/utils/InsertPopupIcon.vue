<template>
  <div class="editor-insert-popup__icon" :class="{ 'is-active': active }">
    <div class="editor-insert-popup__iconContent" v-html="src" />
  </div>
</template>

<script lang="ts">
export default {
  props: {
    active: Boolean,
    src: String
  }
}
</script>

<style scoped>
.editor-insert-popup__icon {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.15s ease-out;
  cursor: pointer;
}

.editor-insert-popup__iconContent {
  transform: scale(1);
}

.editor-insert-popup__icon .fill-target {
  fill: #fff;
}

.editor-insert-popup__icon.is-active .fill-target,
.editor-insert-popup__icon:hover .fill-target {
  fill: #848cd9;
}
</style>
