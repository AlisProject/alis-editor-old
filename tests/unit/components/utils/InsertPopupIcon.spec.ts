import { mount } from '@vue/test-utils'
import InsertPopupIcon from '@/components/utils/InsertPopupIcon.vue'

describe('InsertPopupIcon.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(InsertPopupIcon)
    expect(wrapper.element).toMatchSnapshot()
  })
})
