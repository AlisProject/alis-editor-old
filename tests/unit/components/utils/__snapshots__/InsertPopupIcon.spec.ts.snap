// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsertPopupIcon.vue renders correctly 1`] = `
<div
  class="editor-insert-button__icon"
>
  <div
    class="editor-insert-button__iconContent"
  />
</div>
`;
